tests/common/devices.py

```diff
@@ -750,6 +750,135 @@ def get_feature_status(self):
             feature_status[r[0]] = r[1]
         return feature_status, True
 
+    def _parse_column_positions(self, sep_line, sep_char='-'):
+        """Parse the position of each columns in the command output
+
+        Args:
+            sep_line: The output line separating actual data and column headers
+            sep_char: The character used in separation line. Defaults to '-'.
+
+        Returns:
+            Returns a list. Each item is a tuple with two elements. The first element is start position of a column. The
+            second element is the end position of the column.
+        """
+        prev = ' ',
+        positions = []
+        for pos, char in enumerate(sep_line + ' '):
+            if char == sep_char:
+                if char != prev:
+                    left = pos
+            else:
+                if char != prev:
+                    right = pos
+                    positions.append((left, right))
+            prev = char
+        return positions
+
+
+    def _parse_show(self, output_lines):
+
+        result = []
+
+        sep_line_pattern = re.compile(r"^( *-+ *)+$")
+        sep_line_found = False
+        for idx, line in enumerate(output_lines):
+            if sep_line_pattern.match(line):
+                sep_line_found = True
+                header_line = output_lines[idx-1]
+                sep_line = output_lines[idx]
+                content_lines = output_lines[idx+1:]
+                break
+
+        if not sep_line_found:
+            logging.error('Failed to find separation line in the show command output')
+            return result
+
+        try:
+            positions = self._parse_column_positions(sep_line)
+        except Exception as e:
+            logging.error('Possibly bad command output, exception: {}'.format(repr(e)))
+            return result
+
+        for content_line in content_lines:
+            item = {}
+            for (left, right) in positions:
+                k = header_line[left:right].strip().lower()
```

Comment:
Is it better to preprocess header lines before this nested loop?

Reply:
Sure, updated.